src/models/supplemental_constructors.jl

```diff
@@ -57,8 +57,8 @@ function Bus(number, name, bustype::String, angle, voltage, voltagelimits, basev
 end
 
 """Allows construction of a reserve from an iterator."""
-function StaticReserve(name, contributingdevices::IS.FlattenIteratorWrapper,
-                       timeframe, requirement, _forecasts, InfrastructureSystemsInternal)
-    return StaticReserve(name, collect(contributingdevices),
+function VariableReserve(name, contributingdevices::IS.FlattenIteratorWrapper{D},
```

Comment:
I don't think you need to parameterize this.  Even if you do, since you aren't using D anywhere, you can just write ```FlattenIteratorWrapper{Device}```